lib/srv/db/mysql/engine.go

```diff
@@ -330,3 +356,55 @@ func (e *Engine) makeAcquireSemaphoreConfig(sessionCtx *common.Session) services
 		},
 	}
 }
+
+// appendGCPClientCert calls the GCP API to generate an ephemeral certificate
+// when the instance's RequireSsl setting is true. Return requireSSL so the
+// caller can configure a custom dialer.
+func (e *Engine) appendGCPClientCert(ctx context.Context, sessionCtx *common.Session, tlsConfig *tls.Config) (requireSSL bool, err error) {
```

Comment:
Nit: I've been prefixing methods that do or don't do something depending on some conditions with "maybe" so this method would've been called `maybeAppendGCPClientCert`. Don't know if it's a good habit or not, but `appendGCPClientCert` makes it sounds like it always does that.

Reply:
I changed this multiple times because I didn't like any of the options. I was attempting to limit the number of times we call `GetGCPSQLAdminClient` since it acquires a read lock while also reducing the noise in `connect`. It is also called from `GetCloudSQLPassword`, so breaking the flow to include an `isSSLRequired` means we'd be acquiring the read lock 3 times, which seemed excessive. 

WDYT about calling `GetGCPSQLAdminClient` in `connect` and passing the client to `isSSLRequired` and `appendGCPClientCert` helper functions? The trade-off is the `connect` function becomes a few lines longer while the names of the helper functions are more explicit and don't require `maybe`. Also, a future refactor could remove `GetCloudSQLPassword` from `Auth` and accept the client as a parameter as well so we only have to acquire the client once for all three calls (unless you want me to make that change here as well).

Reply:
I made the changes described in the above comment. Things flow much better. Only trade-off is the connect method is a bit longer.